Amplify/Core/Error/DataError.swift

```diff
@@ -0,0 +1,78 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// Errors associated with data-related operations on Amplify. Commonly
+/// used by the API and DataStore categories.
+enum DataError: Error {
+
+    /// Indicates some data marked as required was unavailable.
+    case dataUnavailable
+
+    /// Indicates that expected data could not be loaded due to the
+    /// lack of integrity.
+    case dataIntegrity(underlyingError: Error? = nil)
+
+    /// Indicates an unexpected scenario happened and the specific
+    /// error could not be resolved.
+    case unknown(ErrorDescription, RecoverySuggestion, Error)
+}
+
+extension DataError: AmplifyError {
```

Comment:
we currently have `CoreError` which is returned on List operations. I think we should merge the two into one, and name it either CoreError or DataError as you have it here